File: 351004/Brazhalovich/Lab_1/src/main/java/org/example/newsapi/dto/request/NewsRequestTo.java
package org.example.newsapi.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import java.util.Set;

@Data
public class NewsRequestTo {
    //@JsonProperty("user") // Тестер будет слать "user", а Jackson запишет это в userId
    @NotNull
    private Long userId;

    @Size(min = 2, max = 64)
    private String title;

    @Size(min = 4, max = 2048)
    private String content;

    //@JsonProperty("marker")
    private Set<Long> markerIds;

    // Джексон увидит этот метод и положит данные из JSON-поля "user" в твой "userId"
    public void setUser(Long user) {
        this.userId = user;
    }

    // Джексон увидит этот метод и положит данные из JSON-поля "marker" в твой "markerIds"
    public void setMarker(java.util.Set<Long> marker) {
        System.out.println(">>> RECEIVED MARKER IDs FOR NEWS: " + marker);
        this.markerIds = marker;
    }
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/dto/news/NewsRequestTo.kt
package com.distcomp.dto.news

import com.distcomp.dto.marker.MarkerRequestTo
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class NewsRequestTo (
    val id: Long? = null,
    @field:NotBlank
    @field:Size(min = 3, max = 64)
    val title: String,
    @field:NotBlank
    @field:Size(min = 4, max = 2048)
    val content: String,
    @field:NotNull
    val userId: Long,
    val markers: List<String>?,
)